nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java

```diff
@@ -356,6 +360,81 @@ public void testListTables() {
     Assertions.assertThat(catalog.tableExists(TABLE_IDENTIFIER)).isTrue();
   }
 
+  private void testRegister(TableIdentifier identifier, String metadataVersionFiles) {
```

Comment:
Could we separate these tests for Nessie specific and general ones?
Could we run the "general" tests on every catalog, and move the `testRegisterTable`, and `testRegisterExistingTable` from `HiveTableTest` to this general place?

Reply:
These test cases are specific to Nessie.

Reply:
If I understand correctly, the PR moves the `HiveCatalog.registerTable` to the `BaseMetastoreCatalog` so it will be available for every `Catalog` which is inheriting from `BaseMetastoreCatalog`. So we should have a test for testing all of the catalogs that implement the `registerTable` that the method is working as expected.
I understand that there are some Nessie specific test cases, but I think the basic functionality should be available and tested for all of the catalogs.

Reply:
I will add the said test-cases and would ping you then, thanks.

Reply:
I think we still miss:
- DynamoDbCatalog
- GlueCatalog

I do not like the duplicated test code, and I think it would be good to have a general Catalog API test for checking the functionality of the new Catalog implementations. Something like `TestCatalog` but for all/most of the catalog API methods. We can do it in a different PR, but it would be good to have them. The test for `registerTable()` is a good candidate for these common tests.

Reply:
I have put in test cases for Glue and DynamoDb as "Ignore" because I currently do not have an AWS account for testing them.
Also, I feel we could address the Catalog API test in a different PR.

Reply:
I got my AWS account setup, and have amended the test cases in Glue and DynamoDb Catalog, @pvary please have a look over the test cases.

Reply:
> I do not like the duplicated test code, and I think it would be good to have a general Catalog API test for checking the functionality of the new Catalog implementations. Something like TestCatalog but for all/most of the catalog API methods. We can do it in a different PR, but it would be good to have them. The test for registerTable() is a good candidate for these common tests.

We do have `CatalogTests`, which is what `TestRESTCatalog` is built off of. It’s nice in that it has some methods for subclasses to declare if the catalog being tested supports certain features, so the tests can exit early if they don’t support an optional feature.

For example if the catalog in question doesn’t support namespace properties.

Might be good to see if that can be made to fit this need or can otherwise get some ideas from it. I believe Nessie also implements these tests.

Reply:
A good suggestion of using `CatalogTests`, but the duplicated test code referred to under these comments above shall not be resolved using `CatalogTests`.
Reason:
1. The duplicated test code is with two of the test cases, i.e., testRegisterTable() and testRegisterExistingTable() in three of the catalog test files.

- Jdbc Catalog
- Ecs Catalog
- Hadoop Catalog

Out of these only Jdbc catalog is one which can use `CatalogTests`, others do not use `CatalogTests`.
Thus, it won't be helpful to consider `CatalogTests` under this comment.